falcon/request.py

```diff
@@ -358,10 +358,23 @@ class Request(object):
             accessed.)
         range_unit (str): Unit of the range parsed from the value of the
             Range header, or ``None`` if the header is missing
-        if_match (str): Value of the If-Match header, or ``None`` if the
-            header is missing.
-        if_none_match (str): Value of the If-None-Match header, or ``None``
-            if the header is missing.
+        if_match (list): A list containing all the etags in the If-Match
+            header.
+
+            This property is determined by the value of ``HTTP_IF_MATCH``
+            in the WSGI environment dict.
+
+            Note:
+                This property includes strong and weakness entity-tags and that
+                means that two entity-tags are equivalent if both are not weak
+                and their opaque-tags match character-by-character.
```

Comment:
As discussed, we would also need to include code recipes, for instance, how to check if the provided strong entity-tags match, and so on.